Possible fix to Sacrificial Lamb and Unholy Alliance

srockw opened the pull request:
Doing this feels very wrong... but it's the only way I've found without having to manually check for every food joker's key. I've tested it and seems to work fine, but I am not too sure if it's a good idea

Would fix #50

OppositeWolf770 commented:
Hmm. It seems to work just fine. I didn't realize that Jokers were included under the `remove_playing_cards` context. I can probably remove that entire custom context I created for these Jokers.

But I guess if it works just fine right now, we might as well implement it in. It's better than not having support for them, and if push comes to shove, we can just revert or modify the changes in the future if issues arise.

srockw commented:
Jokers aren't included in that context, with these changes the logic for playing cards being destroyed is in the `calculate` function (unchanged from how it was, I merely removed the joker part and removed some unnecessary if nesting) while I moved the logic for any joker being destroyed to a `CardArea:remove_card` hook

OppositeWolf770 commented:
Ah, I see. I completely missed the CardArea hook when I was looking over it. Yeah, it has potential to cause issues in the future I suppose, but I guess we'll just have to see. Thanks for this!

srockw commented:
Yeah, the main problem I encountered was that for food jokers getting destroyed, `CardArea:remove_card` is called first, then `Card:remove` is called, hooking into `Card:remove` means it's too late to check whether the joker was actually in the player's joker area, and `Card:remove` is also called for jokers you don't own, like in the shop or even in the collections menu. Add to that that there's no context for this scenario leads to my solution, which, as you said, has potential for future problems. But you're 100% right that it can be reverted later.

Another solution would be to go by the logic that food jokers are "eaten" instead of destroyed and exclude them from these cards.
